src/lazyload-image.ts

```diff
@@ -45,16 +61,34 @@ function loadImage(imagePath: string): Observable<HTMLImageElement> {
         });
 }
 
-function setImage(element: HTMLElement, imagePath: string) {
+function setImage(element: HTMLElement, imagePath: string, useSrcset: boolean, sourcesAttrName: string) {
+    trySetSources(element, sourcesAttrName);
     const isImgNode = element.nodeName.toLowerCase() === 'img';
     if (isImgNode) {
-        (<HTMLImageElement>element).src = imagePath;
+        if (useSrcset) {
+            (<HTMLImageElement>element).srcset = imagePath;
+        } else {
+            (<HTMLImageElement>element).src = imagePath;
+        }
     } else {
         element.style.backgroundImage = `url('${imagePath}')`;
     }
     return element;
 }
 
+function trySetSources(element: HTMLElement, attrName: string) {
+    const isPicture = element.parentElement && element.parentElement.nodeName.toLowerCase() === 'picture';
+    if (isPicture) {
+        const sources = element.parentElement.getElementsByTagName('source');
+        for (let i = 0; i < sources.length; i++) {
```

Comment:
That's the exact reason while I oped-in for `for` loop instead of `forEach`.
However, IE doesn't support srcset attribute by default. More on that in my main post below.
I've left it as it is for now.